pyproject.toml

```diff
@@ -113,6 +113,7 @@ filterwarnings = [
     'error',
     'ignore::asdf.exceptions.AsdfDeprecationWarning:asdf.asdftypes',
     'ignore:numpy.ndarray size changed:RuntimeWarning',
+    'ignore:.*from astropy.io.misc.asdf.types subclasses the deprecated CustomType.*:asdf.exceptions.AsdfDeprecationWarning',
```

Comment:
I think we should filter this specific warning when we load the entry points in `asdf.entry_points`. `astropy.io.misc.asdf` is deprecated and will be removed in `astropy` 6, meaning that `astropy` will not want to resolve this warning.

Since `astropy` includes an entry point which raises this warning, this warning will be raised each time ASDF is used while `astropy` is installed. Normally, I wouldn't care too much about doing this, but since our primary users all use ASDF and `astropy` together this effectively means everything in our downstream will need to suppress this specific warning (in many cases this will be turned into an error), which `astropy` will not be fixing. So we will be breaking our downstream packages' CI unnecessarily.

If we restrict the filter to just `astropy.io.misc.asdf` then this suppression will only effect `astropy` (which is won't be upgrading anyways).